Implementing BigtableTableAdminClietwrapper

rahulKQL opened the pull request:
## What changes it contains
Initial Implementation for `BigtableTableAdminClientwrapper` to adapt to v2.models classes.
This is related to the ongoing wrapping of native cloud BigTable adapters on top of Google-Cloud-java adapter.

-  `AbstractBigtableAdmin`  & `BigtableAdmin`  is now adapting to v2.models, (except i.e. **`getTable()`** & snapshot related operations ).
- Modified `BigtableAdminWrapper#dropRowRange` & `BigtableAdminWrapper#dropRowRangeAsync` to support for truncate and deleteByPrefix.
- `hbase2.xBigtableTableAdminClient` is uneffected by this PR as it contains snapshot related operation (this would be separate PR) .

***
**This PR is only for review**

rahulKQL committed:
Implementing BigtableTableadminClietwrapper

rahulKQL committed:
Implementing BigtableTableadminClietwrapper

rahulKQL commented on bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java:
```diff
@@ -353,10 +345,10 @@ public void modifyTable(TableName tableName, TableDescriptor tableDescriptor) th
-        bigtableTableAdminClient
-          .dropRowRangeAsync(deleteRequest.setName(toBigtableName(tableName)).build()))
+        tableAdminClientWrapper
+          .dropRowRangeAsync(tableName.getNameAsString(), ByteString.EMPTY.toStringUtf8()))
```
Should I use the String `""` (empty quote) here or using ByteString.Empty is better?

romlogic commented on bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableTableAdminClientWrapper.java:
```diff
@@ -196,23 +197,34 @@ public Table apply(com.google.bigtable.admin.v2.Table tableProto) {
   @Override
   public void dropRowRange(String tableId, String rowKeyPrefix) {
-    DropRowRangeRequest requestProto = DropRowRangeRequest.newBuilder()
+    DropRowRangeRequest.Builder dropRequestProtoBuiler = DropRowRangeRequest.newBuilder()
```
is the dropRequestProtoBuiler intentional?

romlogic replied:
I meant why not dropRequestProtoBuilder?

rahulKQL replied:
Thanks!!

rahulKQL committed:
Add feedback changes

rahulKQL committed:
Add feedback changes

sduskis committed:
Removing a superfluous Beam config variable (#2052)

`CloudBigtableScanConfiguration.Builder.scan` was more or less a copy of `CloudBigtableScanConfiguration.Builder.request`.

sduskis committed:
Removing RequestWithKeysValueProvider (#2051)

`CloudBigtableScanConfiguration.withKeys()` is only called after the request materializes.  Adding some `Precondition`s and removing `RequestWithKeysValueProvider` which only added complexity.

sduskis committed:
Minimal addition of Query configuration for Beam (#2050)

* Minimal addition of Query configuration for Beam

* Fixing compilation isues, adding a test.

sduskis committed:
Upgrading google-cloud-clients to 0.77 (#2049)

This upgrades grpc to 0.17.1, and opencensus to 0.17.0

rahulKQL committed:
Rebase and updated

rahulKQL committed:
refactor dropRowRangeRequest in adminWrapper
